ecommerce/src/main/java/com/algolia/instantsearch/examples/ecommerce/EcommerceActivity.java

```diff
@@ -79,6 +99,70 @@ protected void onDestroy() {
         refWatcher.watch(findViewById(R.id.hits));
     }
 
+    // endregion
+
+    // region Permission handling
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
+        if (requestCode == ID_REQ_VOICE_PERM && grantResults.length > 0) {
+            if (grantResults[0] == PackageManager.PERMISSION_GRANTED) {
```

Comment:
To avoid more nesting, I would put all this with the `if` above and `&&`

Reply:
After offline discussion